test/prototype/models/test_transforms.py

```diff
@@ -0,0 +1,58 @@
+import torch
+from test.common.assets import get_asset_path
+from test.common.torchtext_test_case import TorchtextTestCase
+from torchtext.prototype.models import T5Transform
+
+
+class TestTransforms(TorchtextTestCase):
+    def _t5tokenizer(self, test_scripting):
+        asset_name = "t5_tokenizer_base.model"
+        asset_path = get_asset_path(asset_name)
+        transform = T5Transform(asset_path, max_seq_len=512, eos_idx=1, padding_idx=0)
+        if test_scripting:
+            transform = torch.jit.script(transform)
+
+        # input is a list (i.e. batched)
+        input_seq = ["Hello World!, how are you?"]
+        actual = transform(input_seq)
+        expected = torch.tensor([[8774, 1150, 55, 6, 149, 33, 25, 58, 1]])
+        self.assertEqual(actual, expected)
+
+        # input is a single string
+        input_seq = "Hello World!, how are you?"
+        actual = transform(input_seq)
+        expected = torch.tensor([8774, 1150, 55, 6, 149, 33, 25, 58, 1])
+        self.assertEqual(actual, expected)
+
+    def test_t5tokenizer(self):
+        """test tokenization on string input (batched and un-batched)"""
+        self._t5tokenizer(test_scripting=False)
+
+    def test_t5tokenizer_jit(self):
+        """test tokenization with scripting on string input (batched and un-batched)"""
+        self._t5tokenizer(test_scripting=True)
+
+    def _t5tokenizer_decode(self, test_scripting):
```

Comment:
nit: I wonder if we just have both `decode` and `encode` methods tested in the same function i.e `_t5tokeniser`?